File: domain/src/main/java/mixter/domain/core/subscription/handlers/UpdateFollowers.java
package mixter.domain.core.subscription.handlers;

import mixter.doc.Handler;
import mixter.domain.core.subscription.FollowerRepository;
import mixter.domain.core.subscription.events.UserFollowed;

@Handler
public class UpdateFollowers {
    private FollowerRepository repository;

    public UpdateFollowers(FollowerRepository repository) {
        this.repository = repository;
    }

    public void apply(UserFollowed event) {
        repository.saveFollower(event.getSubscriptionId().getFollowee(), event.getSubscriptionId().getFollower());
    }
}

File: domain/src/main/scala/mixter/domain/subscription/handlers/UpdateFollowers.scala
package mixter.domain.subscription.handlers

import mixter.domain.subscription.FollowerRepository
import mixter.domain.subscription.event.UserFollowed

class UpdateFollowers(repository: FollowerRepository) {
  def apply(event:UserFollowed):Unit ={
    repository.saveFollower(event.subscriptionId.followee, event.subscriptionId.follower)
  }
}
